libraries/helpers.rb

```diff
@@ -270,27 +270,49 @@ def installer
       @installer ||= begin
         ensure_mixlib_install_gem_installed!
 
+        resolve_platform!
+
         options = {
           product_name: new_resource.product_name,
           channel: new_resource.channel,
           product_version: new_resource.version,
+          platform: new_resource.platform,
+          platform_version: new_resource.platform_version,
+          architecture: new_resource.architecture,
         }.tap do |opt|
           if new_resource.platform_version_compatibility_mode
             opt[:platform_version_compatibility_mode] = new_resource.platform_version_compatibility_mode
           end
           opt[:shell_type] = :ps1 if windows?
         end
 
-        platform_details = {
-          platform: platform_family_shortname,
-          platform_version: truncate_platform_version(new_resource.platform_version, new_resource.platform),
-          architecture: Mixlib::Install::Util.normalize_architecture(new_resource.architecture),
-        }
+        Mixlib::Install.new(options)
+      end
+    end
 
-        options.merge!(platform_details)
+    def resolve_platform!
+      # Auto detect platform
+      detected_platform = Mixlib::Install.detect_platform
 
-        Mixlib::Install.new(options)
+      if new_resource.platform.nil?
+        # set detected suse platforms to sles for artifact discovery
+        if detected_platform[:platform] =~ /suse/
+          new_resource.platform('sles')
```

Comment:
We should remove this for now as it won't do the right thing in 80% of our products (the ones that don't have native `sles` builds yet. Let's just allow two cases at the moment: 
* user provided platform
* mixlib-install auto-detected platform